### PR TITLE
[OPIK-314] Create prompt version endpoint

### PR DESCRIPTION
**Pull request by thiagohora**
## Details
- Add Create prompt version endpoint

## Issues
OPIK-314

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/domain/PromptService.java [hunk 1]
```diff
@@ -129,9 +137,124 @@ public PromptPage find(String name, int page, int size) {
...
+    }
+
+    private Set<String> getVariables(String template) {
+        return MustacheVariableExtractor.extractVariables(template);
+    }
+
     private EntityAlreadyExistsException newConflict(String alreadyExists) {
         log.info(alreadyExists);
-        return new EntityAlreadyExistsException(new ErrorMessage(alreadyExists));
+        return new EntityAlreadyExistsException(new ErrorMessage(409, alreadyExists));
```
Comment (idoberko2) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/PromptService.java:257]:
```suggestion
        return new EntityAlreadyExistsException(new ErrorMessage(HttpStatus.SC_CONFLICT, alreadyExists));
```

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/PromptResourceTest.java [hunk 2]
```diff
@@ -85,6 +88,14 @@ class PromptResourceTest {
 
     private static final WireMockUtils.WireMockRuntime wireMock;
     public static final String[] IGNORED_FIELDS = {"versionCount", "latestVersion", "template"};
+    public static final String TEMPLATE = """
```
Comment (idoberko2) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/PromptResourceTest.java:91]: Any reason to introduce this duplicate and not use `PromptVersionManufacturer.TEMPLATE`?

**Review by idoberko2 — APPROVED** [review]
Great work! Left two minor comments

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/PromptResourceTest.java (hunk 2 above)
Comment (thiagohora) [reply apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/PromptResourceTest.java:91]: No, I probably forgot to remove it.

File: apps/opik-backend/src/main/java/com/comet/opik/domain/PromptService.java (hunk 1 above)
Comment (thiagohora) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/PromptService.java:257]: I will address both comments in the following PR once I rebase it

**Pull request merged**